src/components/EmployeeEdit/EmployeeEdit.jsx

```diff
@@ -0,0 +1,230 @@
+import React, {useEffect, useState} from 'react';
+import PropTypes from 'prop-types';
+import {Redirect} from 'react-router-dom';
+
+import {makeStyles} from '@material-ui/core/styles';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import Paper from '@material-ui/core/Paper';
+import Divider from '@material-ui/core/Divider';
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    width: `calc(100% - 50px)`,
+    marginTop: theme.spacing(3),
+    overflowX: 'auto',
+    marginLeft: theme.spacing(2),
+    display: 'flex',
+    justifyContent: 'center'
+  },
+  paper: {
+    width: '100%',
+    marginBottom: theme.spacing(2),
+    marginLeft: 70,
+    marginRight: 70
+  },
+  formWrapper: {
+    overflowX: 'auto',
+    marginLeft: theme.spacing(3),
+    marginRight: theme.spacing(3)
+  },
+  title: {
+    padding: theme.spacing(3),
+    textAlign: 'left',
+    fontWeight: 600,
+    fontSize: 20
+  },
+  formGroup: {
+    textAlign: 'left',
+    display: 'flex'
+  },
+  label: {
+    alignItems: 'flex-end',
+    marginBottom: 10,
+    display: 'flex',
+    width: 150
+  },
+  input: {
+    width: 420
+  },
+  formGroupButton: {
+    textAlign: 'right'
+  },
+  container: {
+    display: 'block'
+  },
+  button: {
+    margin: theme.spacing(2),
+    marginLeft: 0
+  },
+  textField: {
+    marginLeft: theme.spacing(1),
+    marginRight: theme.spacing(1),
+    width: 400
+  },
+  errorMessage: {
+    fontSize: 12,
+    color: 'red',
+    alignItems: 'flex-end',
+    marginBottom: 15,
+    display: 'flex'
+  }
+}));
+
+const EmployeeEdit = props => {
+  const initialEmployeeData = {
+    FirstName: '',
+    LastName: '',
+    Title: '',
+    Company: ''
+  };
+  const classes = useStyles();
+  const [cancelClicked, setCancelClicked] = useState(false);
+  const [employee, setEmployee] = useState(initialEmployeeData);
+  const [errorMessage, setErrorMessage] = useState({});
+
+  useEffect(() => {
+    if (props.action === 'edit' && props.employee) {
+      setEmployee(props.employee);
+    }
+  }, [props.action, props.employee, employee]);
```

Comment:
Why would we need `employee` as dependency?

Reply:
console was giving me warning messages, to add the employee dependency